### PR TITLE
Fix edge cases in mouse action tracking code

### LINKED ISSUE #2431: [walking/running] Stutter stop regression
Stutter stop regression since f4897c31396f9e36416d326931b15173a2d2aa17.

https://user-images.githubusercontent.com/45902449/126217656-c67998f1-c8e1-4a05-8c83-a106857a8f49.mp4

I also got the impression that in some situations the hero stops walking in the dungeon even though the left mouse button is held down. This is likely to happen when pointing to impassable areas.

### PR DESCRIPTION
**Pull request by AJenbo**
Fixes #2431

Also:
- Track spell targets
- Merge repeat walk in to the new repeat action code
- Avoid time based repeats (use action frame instead)

### CODE REVIEW

File: Source/controls/plrctrls.cpp [hunk 1]
```diff
@@ -1085,7 +1072,7 @@ void WalkInDir(int playerId, AxisDirection dir)
 	const Direction pdir = FaceDir[static_cast<std::size_t>(dir.x)][static_cast<std::size_t>(dir.y)];
 	const auto delta = player.position.future + pdir;
 
-	if (CanChangeDirection(player))
+	if (player.CanChangeAction())
```
Comment (StephenCWills) [inline Source/controls/plrctrls.cpp:1075]:
```suggestion
	if (!player.IsWalking() && player.CanChangeAction())
```

**Review by StephenCWills — CHANGES_REQUESTED** [review]
The new condition introduced when refactoring `CanChangeDirection()` messed up the camera on controller again. Everything else looks good to me.

**Comment by AJenbo** [comment]
> The new condition introduced when refactoring `CanChangeDirection()` messed up the camera on controller again. Everything else looks good to me.

Thanks for testing.

**Review by StephenCWills — APPROVED** [review]

**Comment by Chance4us** [comment]
Thanks a lot.

**Pull request merged**